### DISCUSSION TITLE
Executing code after starting the app

### DISCUSSION THREAD

**Discussion by robinmuhia** [fastapi/fastapi#9511] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FASTAPI()

app.get('/')
def start():
return "Hello World"

def num_print(num:int):
print(num)

num_print(5)
```

### Description

Start the fastapi server
Then execute the num_print function

In practise, the above would run the num_print function then start the server

### Operating System

Windows

### FastAPI Version

0.92.0

### Python Version

3.11

### Additional Context

Is there anyway that one can write some code to execute after the fastapi app starts? Lifetime dependencies run before startup or after shutdown. Is there a way for us to write code that would execute after the fastapi app starts? For example , in the example above, I would like for my code to first run and spin the server up then run the second function to print number 5

**Comment by jgould22** [comment]
If you are using Gunicorn you can use the server hooks here.

https://docs.gunicorn.org/en/stable/settings.html#server-hooks

**Reply by robinmuhia** [reply]
Thank you for your response. I am currently in development. Would it be advisable to use Gunicorn in development?

**Comment by Kludex** [comment]
Why do you want to do that?

**Reply by robinmuhia** [reply]
My project requires a get request to be executed infinitely after startup to the same api I am building with FASTAPI. There are other solutions i could pivot to but that would require re-writing significant parts of the code base.

**Comment by YuriiMotov** [comment, marked as answer]
Use [lifespan](https://fastapi.tiangolo.com/advanced/events/) to run separate thread and execute requests from that thread

**Reply by robinmuhia** [reply]
Thanks for the reply. If I remember correctly, I used the @onevent startup decorator, but I just saw that it was deprecated for lifespan.
